### DISCUSSION TITLE
ODMR trigger channel

### DISCUSSION THREAD

**Discussion by qku** [Ulm-IQO/qudi-core#41] [Q&A] [answered]
Hello,
I am new to setting up CW ODMR with the new core. For synchronization of counts vs. set frequency, there is a trigger output from the NIDAQ going to the MW signal generator, right? How do I define/know which channel of the NIDAQ is responsible for this? I could not find a corresponding config option in the docfile for setting up ODMR. Thank you in advance :)

**Comment by Neverhorst** [comment, marked as answer]
Hi,
for the NI X-series cards, there is a `ConfigOption` called `sample_clock_output` in the hardware module [`qudi.hardware.ni_x_series.ni_x_series_finite_sampling_input`](https://github.com/Ulm-IQO/qudi-iqo-modules/blob/017841055f0281b7d2cf4633401109a85ed679c2/src/qudi/hardware/ni_x_series/ni_x_series_finite_sampling_input.py). Here you can specify the channel you want to feed via coax cable to your microwave generator.
See also [this line](https://github.com/Ulm-IQO/qudi-iqo-modules/blob/017841055f0281b7d2cf4633401109a85ed679c2/docs/setup_odmr.md?plain=1#L64) in the example config of the ODMR documentation.

**Reply by qku** [reply]
Thank you @Neverhorst, this answers my question. Do you happen to have any experience with the Stanford Research Systems signal generators? We just discovered that they only support triggering via GPIB, and not via a simple BNC trigger :(

**Reply by Neverhorst** [reply]
Personally, I do not have any experience with this device. A quick glance at the manual seems to hint towards using the frequency modulation functionality. If you can not trigger the FM by an external TTL trigger, you probably would need to reverse the roles of NI card and signal generator so that the signal generator is triggering the acquisition of the next sample in the NI card (and not vice versa as it is now). There is probably a smart way to incorporate both possibilities in the code (e.g. via ConfigOptions) if you want to go down that road.
Seems unlikely to me that you can use the current ODMR hardware modules with your device without doing some code changes.

**Reply by qku** [reply]
Thank you @Neverhorst. We eventually decided to build an Arduino-based GPIB controller based on an [open source project](https://github.com/Twilight-Logic/AR488) and used one of the remaining digital inputs of the Arduino as a TTL trigger. It works now, but it was a lot of overhead work just to get CW ODMR working. I can therefore not recommend the SI signal generators for use with qudi.
